Sprint-3/quote-generator/quotes.js

```diff
@@ -1,3 +1,55 @@
+let isAutoQuoteGenerator=false;
+let autoQuoteInterval = null;
+function newQuote(){
+  //Initialising the quote generator function when the page load for the first time
+  quoteGenerator()
+
+  //adding event listener to the new-quote button to generate a new quote when clicked
+  document.getElementById('new-quote').addEventListener('click', function(){
+    quoteGenerator();
+  });
+
+  //auto generate function when auto-generator button is clicked
+  document.getElementById('auto-generator').addEventListener('click', function(){
+    autoQuoteGenerator();
+  });
+}
+
+function autoQuoteGenerator(){
+  //we want three things to immediately happen when the auto-generator button is clicked
+  //1. We clear the interval
+  clearInterval(autoQuoteInterval)
+
+  //2. we toggle auto generator 
+  isAutoQuoteGenerator=!isAutoQuoteGenerator
+
+  //3. Update the text of the auto-generator button to "Auto-Generate is ON" of "OFF" depending on autoQuoteGenerator to be true of false immediately
+  if(isAutoQuoteGenerator){
+    document.getElementById('auto-generator').innerText="Auto-Generate is ON";
+    autoQuoteInterval =setInterval(() => {   
+        quoteGenerator();   
+    }, 15000);
+    }
+    else{
+      document.getElementById('auto-generator').innerText="Auto-Generate is OFF";
+    }
+  
+}
+//updating the quote and author paragraph with the current author and quote
+function quoteGenerator(){
+  //Removing h1 text
+    document.querySelector('h1').innerText='';
+
+    //Storing a randomly chosen quote from the list
+    const quote = pickFromArray(quotes);
+
+    //Updating quote and author paragraph with the quote key-value pairs
+    document.getElementById('quote').innerHTML ='<span id="quotation-mark">"</span>'+quote.quote;
+    document.getElementById('author').innerText = "- "+quote.author;
+
+  }
```

Comment:
please  don't try to use innerHTML unless there is no other option. In order to  understand why we should hesitate to use innerHTML you can check this article https://medium.com/@sparklewebhelp/understanding-the-risks-of-using-inner-html-in-web-development-30d4fa67f815